libsolidity/parsing/Parser.cpp

```diff
@@ -333,9 +336,17 @@ Parser::FunctionHeaderParserResult Parser::parseFunctionHeader(bool _forceEmptyN
 {
 	RecursionGuard recursionGuard(*this);
 	FunctionHeaderParserResult result;
-	expectToken(Token::Function);
-	if (_forceEmptyName || m_scanner->currentToken() == Token::LParen)
-		result.name = make_shared<ASTString>(); // anonymous function
+
+	if (m_scanner->currentToken() == Token::Function)
+		result.isConstructor = false;
```

Comment:
Please add a comment that this might be set to true later for old style constructors.